### ISSUE TITLE
Event with empty array rejected?

### ISSUE THREAD

**Issue by jsmorph** [timbray/quamina#140]
I'm getting the error "at line 1 col 6: illegal character ] in array" when asking `MatchesForEvent` with an event that has a property value that's a zero-length array. Seems like `[]` should be a legal value.

Example in the [playground](https://go.dev/play/p/z24ifu1TijC):

```Go
func main() {
	event := `{"a":[]}`
	q, _ := quamina.New()
	if _, err := q.MatchesForEvent([]byte(event)); err != nil {
		log.Fatal(err)
	}
}
```

Output:

```
2009/11/10 23:00:00 at line 1 col 6: illegal character ] in array
```

I'll try to dig in.

**Comment by timbray** [comment]
Yep, looks like a bug to me.  The state machine in flattenJSON; should be easy to fix, I can if you don't feel like it.   But the flattener will return no fields, because when you have an array value, it returns one field for each value. If there are zero, well, zero.  So there's no way to write a Quamina rule to match this.  Except for {"a": [ {"exists":true} ] }

**Comment by jsmorph** [comment]
I recall I was going to gather up some code for random event and pattern generation. I should get to that!

**Comment by jsmorph** [comment]
@timbray If it's an easy fix for you, then that'd be great. If not, I'll take a look. What little I knew about flatterns has evaporated.

**Comment by yosiat** [comment]
Looks like this https://github.com/timbray/quamina/pull/83 PR already fixed the issue and it wasn't released.
If you do `go get github.com/timbray/quamina@main` you won't have any errors.

@timbray maybe worth doing a release for this fix & other changes done?

**Comment by timbray** [comment]
I'd like to get the exists:false bug fix also into the next release. Let's see how review on that goes, if it's slow then sure, we can go ahead and do a release soon.

**Comment by timbray** [comment]
Fixed in b04e6d5febd0b15b8ca90d15e7f8e3d477c1a22e
